aopy/precondition/eye.py

```diff
@@ -0,0 +1,117 @@
+                    num_sd=1.5, intersaccade_min=0.02, max_saccade_duration=0.15, 
+                    debug=False, debug_window=(0, 2)):
+    '''
+    Detect saccades from eye position data. Starts with lowpass filtering of the position, then thresholding
+    of the acceleration, ensuring that saccades are spaced by some minimum intersaccade time, and finally 
+    only considering events with both positive and negative crossings within the minimum saccade time. Optional
+    plots can be generated showing threshold crossings in position, velocity, and acceleration.
+
+    Example:
+
+        Detecting saccades on the first 5 seconds of test data from beignet block 5974 (included in /tests/data/beignet)
+    
+        .. image:: _images/detect_saccades.png
+        .. image:: _images/detect_saccades_hist.png
+        .. image:: _images/detect_saccades_scatter.png
+    
+    Args:
+        eye_pos (nt, nch): eye position data, e.g. from oculomatic
+        samplerate (float): sampling rate of the eye data
+        downsamplerate (float, optional): sampling rate at which to do saccade analysis 
+        lowpass_freq (float, optional): low cutoff frequency to limit analysis of saccades
+        taper_len (float, optional): length of tapers to use in multitaper lowpass filter
+        num_sd (float, optional): number of standard deviations above zero to threshold acceleration
+        intersaccade_min (float, optional): minimum time (in seconds) allowed between saccade onsets
+        max_saccade_duration (float, optional): maximum time (in seconds) that a saccade can take
+        debug (bool, optional): if True, display a figure showing the threshold crossings
+        debug_window (tuple, optional): (start, end) time (in seconds) for the debug figure
+        
+    Returns:
+        tuple: tuple containing:
+        | **onset (nsaccade):** onset time (in seconds) of each detected saccade
+        | **duration (nsaccade):** duration (in seconds) of each detected saccade
+        | **distance (nsaccade):** distance (same units as eye_pos) of each detected saccade
+    '''
+    # Downsample
+    if samplerate > downsamplerate:
+        if debug:
+            print(f"downsampling from {samplerate} to {downsamplerate}")
+        eye_pos = downsample(eye_pos, samplerate, downsamplerate)
+        samplerate = downsamplerate
+
+    # Lowpass filter
+    n_pad = int(1.0*samplerate)
+    eye_pos = np.pad(eye_pos, ((n_pad,n_pad),(0,0)), mode='edge')
+    eye_pos = mt_lowpass_filter(eye_pos, lowpass_freq, taper_len, samplerate)
+    eye_pos = eye_pos[n_pad:-n_pad,:]
+    
+    # Differentiate twice to get acceleration
+    time = np.arange(eye_pos.shape[0])/samplerate
+    velocity = derivative(time, eye_pos, norm=True)
+    accel = derivative(time, velocity, norm=False)
+
+    # Set an appropritate threshold to detect saccades
+    baseline_mean = np.mean(accel)
+    baseline_std = np.std(accel)
```

Comment:
How about setting threshold as the argument? The current code computes the threshold at each trial, but if you want to compare the number of saccades across trials and days, I thought it's better to change this code so that we can set the same threshold across trials and days. I don't know how much changes in baseline there is across trials and days , but there might be big changes in baseline before and after learning in difficult tasks such as BMI control.